src/Entity/ProductsAwareInterface.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Loevgaard\SyliusBrandPlugin\Entity;
+
+use Doctrine\Common\Collections\Collection;
+use Sylius\Component\Core\Model\ProductInterface;
+
+interface ProductsAwareInterface
+{
+    public function initializeProductsCollection(): void;
+
+    /**
+     * @return Collection|BrandAwareInterface[]|ProductInterface[]
+     */
+    public function getProducts(): Collection;
+
+    /**
+     * @param BrandAwareInterface $product
+     * @return bool
+     */
+    public function hasProduct(BrandAwareInterface $product): bool;
+
+    /**
+     * @param BrandAwareInterface $product
+     */
+    public function addProduct(BrandAwareInterface $product): void;
+
+    /**
+     * @param BrandAwareInterface $product
+     */
+    public function removeProduct(BrandAwareInterface $product): void;
```

Comment:
Happy new year @Prometee! Following our latest discussions, shouldn't the methods here have `ProductInterface` as type hint? The `BrandAwareInterface` is not by definition a product therefore I find it misleading that you can i.e.   do `removeProduct($notAproductButImplementsBrandAwareInterface)` if you understand what I mean :)

Reply:
I understand your point but this will not pass the PHPStan checks, because like I say before, Traits must keep the smallest signature.
There is the PHPStan errors while I try to change signatures :
```
------ ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Entity/ProductsAwareInterface.php                                                                                                                                                                                                  
 ------ ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  23     PHPDoc tag @param for parameter $product with type Loevgaard\SyliusBrandPlugin\Entity\BrandAwareInterface|Sylius\Component\Core\Model\ProductInterface is not subtype of native type Sylius\Component\Core\Model\ProductInterface  
  28     PHPDoc tag @param for parameter $product with type Loevgaard\SyliusBrandPlugin\Entity\BrandAwareInterface|Sylius\Component\Core\Model\ProductInterface is not subtype of native type Sylius\Component\Core\Model\ProductInterface  
  33     PHPDoc tag @param for parameter $product with type Loevgaard\SyliusBrandPlugin\Entity\BrandAwareInterface|Sylius\Component\Core\Model\ProductInterface is not subtype of native type Sylius\Component\Core\Model\ProductInterface  
 ------ -----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

 ------ ---------------------------------------------------------------------------------------------- 
  Line   Entity/ProductsAwareTrait.php (in context of class Loevgaard\SyliusBrandPlugin\Entity\Brand)  
 ------ ---------------------------------------------------------------------------------------------- 
  46     Call to an undefined method Sylius\Component\Core\Model\ProductInterface::setBrand().         
  57     Call to an undefined method Sylius\Component\Core\Model\ProductInterface::setBrand().         
 ------ ---------------------------------------------------------------------------------------------- 
``` 
If I remove BrandAwareInterface from the PHPDoc blocks, the first 3 errors disappear but the last two are still there because we need to know that `$product` has a `setBrand()` method.